src/Sylius/Bundle/AdminBundle/Controller/RemoveCatalogPromotionController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Processor\CatalogPromotionRemovalProcessorInterface;
+use Sylius\Component\Promotion\Exception\CatalogPromotionNotFoundException;
+use Sylius\Component\Promotion\Exception\InvalidCatalogPromotionStateException;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class RemoveCatalogPromotionController
+{
+    public function __construct(
+        private CatalogPromotionRemovalProcessorInterface $catalogPromotionRemovalProcessor,
+        private Session $session,
```

Comment:
If you actually need this, it would be better to inject the request stack instead, but I'm guessing you could simply take the session off of the current request inside the `__invoke`.